internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -82,6 +85,9 @@ type DeploymentModel struct {
 	Tags          map[string]string `tfschema:"tags"`
 }
 
+func (m *DeploymentModel) isBasic() bool            { return strings.HasPrefix(m.Sku, "basic") }
+func (m *DeploymentModel) hasDefaultCapacity() bool { return m.Capacity == defaultCapacity }
```

Comment:
this is not a typically pattern in the provider, could we just inline these checks please?

Reply:
done